### DISCUSSION TITLE
Logback instrumentation doesn't inject context when an exception is throwed

### DISCUSSION THREAD

**Discussion by dimitri995** [open-telemetry/opentelemetry-java-instrumentation#7236] [Q&A]
Hello everyone,

I am using the logback instrumentation in order to inject context in my logs in order to have a correlation with my traces.

It's working fine when there is no exception throwed. When an exception is throwed the context is not injected.. I am not using the javaagent but when I try with it the result is the same.
Have you ever face this issue ?

![image](https://user-images.githubusercontent.com/66120155/202758116-5e840d9f-8472-41cc-ad3f-896e286bd63d.png)

This is my logback config : 
![image](https://user-images.githubusercontent.com/66120155/202758680-b191d28f-b079-4332-aacd-1ff5cea0fe86.png)

My OtelConfig:
![image](https://user-images.githubusercontent.com/66120155/202758795-67ca50fa-df7a-412c-b17d-aebb34048dfb.png)

Thank you for your time,

Regards,

Dimitri

**Comment by laurit** [comment]
Aren't `Servlet.service() for servlet ...` lines logged by tomcat that doesn't use logback?

**Reply by dimitri995** [reply]
Hello @laurit,

Yes I am using logback as you can see both log.info & log.error in my query is not instrumented...

![image](https://user-images.githubusercontent.com/66120155/203129436-d4168564-ade3-43bc-a7fc-d7dd2966b834.png)

![image](https://user-images.githubusercontent.com/66120155/203129319-978a6b8f-8fdf-43a2-af12-2219c9596253.png)

**Reply by laurit** [reply]
These reactive callbacks are probably called on a different thread form what `getTrace` is called on. To propagate context into these callbacks you'll need to have instrumentation for whatever the http client is that you use and probably also for the reactive framework you use. Alternatively you could capture context in `getTrace` with `Context context = Context.current();` and in the callback activate that context with `try (Scope ignored = context.makeCurrent()) { // do your logging here }`
